### PR TITLE
Amper: Kaitlin Forsman

### PR DESCRIPTION
**Pull request by kcforsman**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | I chose to keep the room class separate, because I found it made the most sense for me to track the booking and availability. I could have found some way to store the information in the reservations or within my User class, and made rooms just a string. I feel that room as a class leaves more opportunity for ease of change in the future.
Describe a concept that you gained more clarity on as you worked on this assignment. | I don't think I implemented it well and truthfully I'm still working on refactoring everything. I feel like while working on this project, I was thinking a lot about dependencies and how to work with them and how to pull them out of code (some). I still need to apply it across the code but I do feel like the project has helped a lot with that concept.
Describe a _nominal test_ that you wrote for this assignment. | reserve_room in User class needs to return a instance of a Reservation, so the test confirms the return is what's expected
Describe an _edge case test_ that you wrote for this assignment. | I wrote what I think is an edge-case for create a Block, so that it can't have less than 2 rooms. If a party were only renting 1 room it wouldn't make sense to be able to make a block rather than a normal single reservation. 
How do you feel you did in writing pseudocode first, then writing the tests and then the code? | I feel that I am very comfortable with writing the tests and then writing the code, but I am struggling a little with what is function pseudocode. I think my version of pseudocode was creating flow charts and outlines for what different classes I wanted and what I felt they should do.

### CODE REVIEW

File: lib/block.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+require 'date'
+require 'pry'
+
+module Hotel
+  class Block < Reservation
+    attr_reader :id, :reservations
+    def initialize(id, rooms, party, date_range, discount)
+      @id = id
```
Comment (CheezItMan) [inline lib/block.rb]: You should be using `super` here.  Nice work using inheritance however!

File: lib/user.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  TOTAL_ROOMS = 20
+  class User
+    attr_reader :rooms
+    def initialize(rooms)
+      if rooms.size != TOTAL_ROOMS
+        raise StandardError.new("incorrect number of rooms")
+      end
+      @rooms = rooms
+      @reservations = []
+      @calendar = {}
```
Comment (CheezItMan) [inline lib/user.rb:15]: Interesting idea having a hash of the reservation dates, very efficient.

**Comment by CheezItMan** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility | Check, Overall nice work here.
Demonstrated loose coupling |  Check
Methods demonstrate a good use of encapsulation, inputs and outputs | Check
Wave 1 requirements |  Check, nicely done
Wave 2 requirements |  Check, well done, I like how you used a hash of dates for reservations, very time, efficient.
Wave 3 requirements |  Check, nice work using inheritance with `Block`.  
Summary | Nicely done, you hit all the learning goals for the project.  Good use of inheritance and hashes for finding bookings.  Check out our [reference implementation](https://github.com/droberts-ada/hotel/tree/dpr/solution)
